packages/injectable/src/dependency-injection-container/createContainer.js

```diff
@@ -282,15 +282,17 @@ export default (...listOfGetRequireContexts) => {
 };
 
 const autoRegisterInjectables = ({ getRequireContextForInjectables, di }) => {
-  const requireContextForInjectables = getRequireContextForInjectables();
-
-  pipeline(
-    requireContextForInjectables,
-    invoke('keys'),
-    map(requireContextForInjectables),
-    map('default'),
-    forEach(di.register),
-  );
+  const context = getRequireContextForInjectables();
+
+  for (const key of context.keys()) {
+    const injectable = context(key).default;
+
+    if (!injectable) {
```

Comment:
Hmm. This would also benefit from some manual run-time type checking. Consider if default export in `.injectable.{js|ts}` is something that does not walk-and-talk like an `injectable`.

That would mean the injectable at least... `conforms({ id: isString, instantiate: isFunction })`.